impl/src/main/java/org/jboss/weld/bootstrap/WeldRuntime.java

```diff
@@ -55,6 +57,8 @@ public BeanManagerImpl getManager(BeanDeploymentArchive beanDeploymentArchive) {
 
     public void shutdown() {
         try {
+            // fire Shutdown event for all modules first
+            fireEventForAllModules(Shutdown.class, new Shutdown(), Any.Literal.INSTANCE);
```

Comment:
Shouldn't there be a guard here?
```suggestion
if (environment.automaticallyHandleStartupShutdownEvents()){
            fireEventForAllModules(Shutdown.class, new Shutdown(), Any.Literal.INSTANCE);
}
```

Reply:
Yeah, you are right. That's what I get for context switching :)
https://github.com/weld/core/pull/2713 fixes it